src/Shared/Components/Security/Vulnerabilities/Vulnerabilities.tsx

```diff
@@ -0,0 +1,152 @@
+import { VulnerabilitiesProps } from './types'
+import { SecuritySummaryCard } from '../SecuritySummaryCard'
+import { getSecurityScan } from '../SecurityModal'
+
+const Vulnerabilities = ({
+    isScanned,
+    isScanEnabled,
+    artifactId,
+    applicationId,
+    environmentId,
+    setVulnerabilityCount,
+    isScanV2Enabled,
+}: VulnerabilitiesProps) => {
+    const [areVulnerabilitiesLoading, vulnerabilitiesResponse, vulnerabilitiesError, reloadVulnerabilities] = useAsync(
+        () => getLastExecutionByArtifactAppEnv(artifactId, applicationId, environmentId),
+        [],
+        isScanned && isScanEnabled && !isScanV2Enabled,
+        {
+            resetOnChange: false,
+        },
+    )
+
+    const [scanResultLoading, scanResultResponse, scanResultError, reloadScanResult] = useAsync(
+        () => getSecurityScan({ artifactId, appId: applicationId, envId: environmentId }),
+        [],
+        isScanned && isScanEnabled && isScanV2Enabled,
+        {
+            resetOnChange: false,
+        },
+    )
+
+    useEffect(() => {
+        if (scanResultResponse && isScanV2Enabled) {
+            setVulnerabilityCount(scanResultResponse.result.imageScan.vulnerability?.list?.length)
+            return
+        }
+        if (vulnerabilitiesResponse && !isScanV2Enabled) {
+            setVulnerabilityCount(vulnerabilitiesResponse.result.vulnerabilities?.length)
+        }
+    }, [vulnerabilitiesResponse, scanResultResponse])
+
+    if (!isScanEnabled) {
+        return (
+            <div className="security-tab-empty">
+                <p className="security-tab-empty__title">Scan is Disabled</p>
+            </div>
+        )
+    }
+
+    if (areVulnerabilitiesLoading || scanResultLoading) {
+        return (
+            <div className="security-tab-empty">
+                <Progressing />
+            </div>
+        )
+    }
+
+    if (
+        !isScanned ||
+        (vulnerabilitiesResponse && !vulnerabilitiesResponse.result.scanned) ||
+        (scanResultResponse && !scanResultResponse?.result.scanned)
+    ) {
+        return (
+            <div className="security-tab-empty">
+                <p className="security-tab-empty__title">Image was not scanned</p>
+            </div>
+        )
+    }
+
+    if (vulnerabilitiesError || scanResultError) {
+        return (
+            <div className="security-tab-empty">
+                <p className="security-tab-empty__title">Failed to fetch vulnerabilities</p>
+                <button
+                    className="cta secondary"
+                    type="button"
+                    onClick={isScanV2Enabled ? reloadScanResult : reloadVulnerabilities}
+                >
+                    Reload
+                </button>
+            </div>
+        )
+    }
+
+    const imageScanSeverities = scanResultResponse?.result.imageScan.vulnerability?.summary.severities
+    const severityCount: SeverityCount = isScanV2Enabled
```

Comment:
Can use a util to get severityCount and totalCount, since also used at 2 other places in dashboard